dht/token.hh

```diff
@@ -42,6 +42,37 @@ enum class token_kind {
     after_all_keys,
 };
 
+struct legacy_token {
+    token_kind _kind;
+    int64_t _data;
+
+    legacy_token(token_kind k, int64_t d) : _kind(k), _data(d) {}
+
+    // IDL constructor.
+    legacy_token(token_kind k, const bytes& b) {
+        switch (k) {
+        case token_kind::before_all_keys:
+            _data = std::numeric_limits<int64_t>::min();
+            break;
+        case token_kind::after_all_keys:
+            // Should not be happening. Throw?
```

Comment:
Why can't this happen? Won't an old node send after_all_keys to a new node, which should translate it to something else? I saw something like this in the ring_position deserializer.

Reply:
The comment was wrong. (It became outdated at some point while I was working on the series, and I forgot to remove it.) Fixed.